students/ZidanLuo/session04/trigrams.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+import sys
+import random
+
+def build_trigrams(words):
+    """
+    build up the trigrams dict from the list of words
+	    return the a dict with:
+	        keys: word pairs
+	        values: list of followers
+    """
+    trigrams = {}
+    for i in range(len(words)-2):
+        pair = words[i:i+2]
+        followers = words[i+2]
+        if tuple(pair) in trigrams.keys():
+            trigrams[tuple(pair)].append(followers)
+        else:
+            trigrams[tuple(pair)] = [followers]
+    return trigrams
+
+def read_in_data(filename):
+    with open(filename, 'r') as f:
+        lines = f.readlines()[61:]
+
+    end_sentence = [line for line in lines if line.startswith("End of the Project Gutenberg EBook")]
+    index = lines.index(end_sentence[0])
```

Comment:
Getting an error when I run this - probably because of the version of sherlock.txt I'm using.  This may be a good spot for an exception :)